cluster-autoscaler/utils/gpu/gpu.go

```diff
@@ -71,19 +93,51 @@ func FilterOutNodesWithUnreadyGpus(allNodes, readyNodes []*apiv1.Node) ([]*apiv1
 // GetGpuTypeForMetrics returns name of the GPU used on the node or empty string if there's no GPU
 // if the GPU type is unknown, "generic" is returned
 // NOTE: current implementation is GKE/GCE-specific
-func GetGpuTypeForMetrics(node *apiv1.Node) string {
+func GetGpuTypeForMetrics(node *apiv1.Node, nodeGroup cloudprovider.NodeGroup, template *schedulercache.NodeInfo) string {
 	// we use the GKE label if there is one
-	gpuType, found := node.Labels[GPULabel]
-	if found {
-		return gpuType
+	gpuType, labelFound := node.Labels[GPULabel]
+	capacity, capacityFound := node.Status.Capacity[ResourceNvidiaGPU]
+
+	// GKE-specific label & capacity are present - consistent state
+	if labelFound && capacityFound {
+		return validateGpuType(gpuType)
+	}
+	// GKE-specific label present but no capacity (yet?) - check the node template
+	if labelFound {
+		if nodeGroup == nil && template == nil {
+			return MetricsErrorGPU
```

Comment:
As discussed offline, let's use another constant to distinguish the cases

Reply:
Done.